src/Pulsar.Client/Internal/ConnectionPool.fs

```diff
@@ -74,18 +107,18 @@ type internal ConnectionPool (config: PulsarClientConfiguration) =
             log("RemoteCertificateNameMismatch")
             result
         | SslPolicyErrors.RemoteCertificateChainErrors ->
-            let result =
-                config.TlsAllowInsecureConnection
-                ||
-                if isNull config.TlsTrustCertificate then
-                    false
-                else
-                    chain.ChainPolicy.ExtraStore.Add(config.TlsTrustCertificate) |> ignore
-                    use cert2 = new X509Certificate2(cert)
-                    chain.Build(cert2)
+            let result = config.TlsAllowInsecureConnection || CheckRemoteCertWithTrustCertificate()
             let log = if result then Log.Logger.LogWarning else Log.Logger.LogError
             log("RemoteCertificateChainErrors")
             result
+        | error when error = SslPolicyErrors.RemoteCertificateChainErrors + SslPolicyErrors.RemoteCertificateNameMismatch ->
+            let result =
+                not config.TlsHostnameVerificationEnable
+                &&
```

Comment:
Can you please move it to the upper line

Reply:
Fixed